src/persist-client/src/iter.rs

```diff
@@ -101,12 +100,15 @@ fn interleave_updates<T: Codec64, D: Codec64>(
 
     let mut arrays: Vec<&dyn Array> = Vec::with_capacity(updates.len());
     let mut interleave = |get_array: fn(&BlobTraceUpdates) -> Option<&dyn Array>| {
+        arrays.clear();
         for part in updates {
             arrays.push(get_array(part)?);
         }
-        let out = ::arrow::compute::interleave(arrays.as_slice(), &indices).ok();
-        arrays.clear();
-        out
+        let out = ::arrow::compute::interleave(arrays.as_slice(), &indices);
+        if let Err(err) = &out {
+            warn!("interleave error: {}", err);
+        }
+        out.ok()
```

Comment:
didn't look too closely, apologies if it's obvious, what was this change for?

Reply:
It was possible before to exit this function without clearing `arrays`, which meant that future calls to the function might see arrays from previous calls. Not great!

This bug is already in main, though I don't know of any way to actually trigger it. (Since codec arrays were always present before.) I'd be most comfortable if we get the fix into this release however!